src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js

```diff
@@ -0,0 +1,165 @@
+import environment from 'platform/utilities/environment';
+import { apiRequest } from 'platform/utilities/api';
+import { getCernerURL } from 'platform/utilities/constants';
+
+export class CernerCallToAction extends Component {
+  static propTypes = {
+    callToActions: PropTypes.arrayOf(
+      PropTypes.shape({
+        deriveHeaderText: PropTypes.func.isRequired,
+        href: PropTypes.string.isRequired,
+        label: PropTypes.string.isRequired,
+      }),
+    ),
+    type: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+    // From mapStateToProps.
+    facilities: PropTypes.arrayOf(
+      PropTypes.shape({
+        facilityId: PropTypes.string.isRequired,
+        isCerner: PropTypes.bool.isRequired,
+      }).isRequired,
+    ).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: '',
+      facilities: [],
+      fetching: true,
+    };
+  }
+
+  componentDidMount() {
+    const { facilities } = this.props;
+
+    // Escape early if there are no facilities.
+    if (isEmpty(facilities)) {
+      return;
+    }
+
+    // Derive the cerner facilities.
+    const cernerFacilities = facilities.filter(facility => facility.isCerner);
+
+    // Escape early if there are no cerner facilities.
+    if (isEmpty(cernerFacilities)) {
+      // WARNING: Add sentry logging here if there are no cerner facilities found, this should never happen as the component only renders when there ARE cerner facilities.
+      return;
+    }
+
+    // Derive the list of facility IDs.
+    const cernerFacilityIDs = map(
+      cernerFacilities,
+      facility => `vha_${facility.facilityId}`,
+    );
+
+    // Fetch cerner facilities.
+    this.fetchFacilities(cernerFacilityIDs);
+  }
+
+  fetchFacilities = async facilityIDs => {
+    // Show loading state.
+    this.setState({ fetching: true });
+
+    try {
+      // Fetch facilities and store them in state.
+      const response = await apiRequest(
+        `${environment.API_URL}/v1/facilities/va?ids=${facilityIDs.join(',')}`,
+      );
+      this.setState({ facilities: response?.data, fetching: false });
+
+      // Log any API errors.
+    } catch (error) {
+      this.setState({ error: error.message, fetching: false });
+    }
+  };
+
+  render() {
+    const { callToActions, text, type } = this.props;
+    const { error, fetching, facilities } = this.state;
+
+    // Escape early if we are fetching.
+    if (fetching) {
+      return <LoadingIndicator message="Loading your information..." />;
+    }
+
+    // Escape early if there was an error fetching the Cerner facilities.
+    if (error || isEmpty(facilities)) {
+      // WARNING: Add sentry logging here if there is an error fetching Cerner facilities.
+      return (
+        <AlertBox
+          headline="Something went wrong"
+          content="We’re sorry. Something went wrong on our end. Please try again later."
```

Comment:
Is this ok messaging for when there's an error fetching the Cerner facility names @DanielleThierryUSDSVA for a user that belongs to a Cerner facility? 🙂

Reply:
Hi @kelsonic. I think it's fine for now for Cerner testing. But thinking more on it...the user hasn't taken any action at this point, right? They've just arrived at the page and we couldn't pull facility names?

Reply:
Yupp, that's correct! I think that might be a little weird for the user since this error wasn't prompted by any user input and the language seems like it was, which is why I wanted to flag it 🙂